hack/tools/release/notes/main.go

```diff
@@ -117,6 +115,26 @@ func (cmd *notesCmd) run() error {
 	return generator.run()
 }
 
+func releaseTypeFromNewTag(newTagConfig string) string {
+	// Error handling can be ignored as the version has been validated in computeConfigDefaults already.
+	newTag, _ := semver.ParseTolerant(newTagConfig)
+
+	// Return early if it is not a pre release.
+	if len(newTag.Pre) != 2 {
+		return ""
+	}
```

Comment:
In this case only versions like `v1.7.0-beta.1` will be handled, but not `v1.7.0-beta` or `v1.7.0-beta1`. I think this is fine, WDYT @willie-yao @cahillsf?

Reply:
Otherwise we can change the code to

```suggestion
	// Return early if it is not a pre release.
	if len(newTag.Pre) == 0 {
		return ""
	}
```

Reply:
Added a comment, PTAL